### ISSUE TITLE
Debug mode breaks login

### ISSUE THREAD

**Issue by SwampFalc** [kanboard/kanboard#2955]
I'm a new user. I've installed everything and it runs fine using the default admin account.

So then I tried to connect it to my LDAP, which didn't seem to work at first sight. I turned on debug mode and it still didn't work AND I didn't really get any useful information.

I did notice that I couldn't login at all, not even with the default admin. I assumed this was because of the broken LDAP config, but after a bit more fiddling, I now believe that debug mode completely blocks logging in. Which is of course very annoying if you need to debug an LDAP connection.

I have a sneaking suspicion #2936 might be related to this...

### Expected behaviour

Logging in with 'admin' should work even in debug mode.

### Actual behaviour

Error messages:

The username is required
The password is required

### Steps to reproduce

1. Clean install
2. Change config to enable debug to file:
  // Enable/Disable debug                                                                                                                                                          define('DEBUG', false);
  // Available log drivers: syslog, stderr, stdout or file                                                                                                                            
  define('LOG_DRIVER', 'file');

3. Try to log in

### Configuration

Copy and paste the configuration section from the Kanboard settings page or fill these fields:

- Kanboard version: 1.0.37
- Database type and version: sqlite (standard version for Debian Jessie)
- PHP version: PHP 5.6.29-0+deb8u1
- OS: Debian Jessie
- Browser: Firefox

**Comment by fguillot** [comment]
Do you have any PHP errors in the web server log files?

**Comment by SwampFalc** [comment]
Good idea, hadn't thought of it yet.

My nginx error log for 1 failed login:

```
2017/01/18 09:16:21 [error] 24855#0: *165 FastCGI sent in stderr: "PHP message: PHP Warning:  file_put_contents(data/debug.log): failed to open stream: No such file or directory in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 44
PHP message: PHP Fatal error:  Uncaught exception 'RuntimeException' with message 'Unable to write to the log file.' in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php:45
Stack trace:
#0 /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/Logger.php(76): SimpleLogger\File->log('debug', 'SQL: SELECT "op...', Array)
#1 /srv/kanban/vendor/psr/log/Psr/Log/AbstractLogger.php(126): SimpleLogger\Logger->log('debug', 'SQL: SELECT "op...', Array)
#2 /srv/kanban/app/Subscriber/BootstrapSubscriber.php(32): Psr\Log\AbstractLogger->debug('SQL: SELECT "op...')
#3 [internal function]: Kanboard\Subscriber\BootstrapSubscriber->__destruct()
#4 {main}
  thrown in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 45" while reading response header from upstream, client: 10.0.4.9, server: kanban.fedasil.be, request: "GET /?controller=AuthController&action=logout HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm.sock:", host: "kanban.fedasil.be"
2017/01/18 09:16:21 [error] 24855#0: *165 FastCGI sent in stderr: "PHP message: PHP Warning:  file_put_contents(data/debug.log): failed to open stream: No such file or directory in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 44
PHP message: PHP Fatal error:  Uncaught exception 'RuntimeException' with message 'Unable to write to the log file.' in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php:45
Stack trace:
#0 /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/Logger.php(76): SimpleLogger\File->log('debug', 'SQL: SELECT "op...', Array)
#1 /srv/kanban/vendor/psr/log/Psr/Log/AbstractLogger.php(126): SimpleLogger\Logger->log('debug', 'SQL: SELECT "op...', Array)
#2 /srv/kanban/app/Subscriber/BootstrapSubscriber.php(32): Psr\Log\AbstractLogger->debug('SQL: SELECT "op...')
#3 [internal function]: Kanboard\Subscriber\BootstrapSubscriber->__destruct()
#4 {main}
  thrown in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 45" while reading upstream, client: 10.0.4.9, server: kanban.fedasil.be, request: "GET /?controller=AuthController&action=login HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm.sock:", host: "kanban.fedasil.be"
2017/01/18 09:16:36 [error] 10723#0: *2 FastCGI sent in stderr: "PHP message: PHP Warning:  file_put_contents(data/debug.log): failed to open stream: No such file or directory in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 44
PHP message: PHP Fatal error:  Uncaught exception 'RuntimeException' with message 'Unable to write to the log file.' in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php:45
Stack trace:
#0 /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/Logger.php(76): SimpleLogger\File->log('debug', 'SQL: SELECT "op...', Array)
#1 /srv/kanban/vendor/psr/log/Psr/Log/AbstractLogger.php(126): SimpleLogger\Logger->log('debug', 'SQL: SELECT "op...', Array)
#2 /srv/kanban/app/Subscriber/BootstrapSubscriber.php(32): Psr\Log\AbstractLogger->debug('SQL: SELECT "op...')
#3 [internal function]: Kanboard\Subscriber\BootstrapSubscriber->__destruct()
#4 {main}
  thrown in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 45" while reading upstream, client: 10.0.4.9, server: kanban.fedasil.be, request: "GET /?controller=AuthController&action=login HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm.sock:", host: "kanban.fedasil.be"
2017/01/18 09:16:41 [error] 10723#0: *2 FastCGI sent in stderr: "PHP message: PHP Warning:  file_put_contents(data/debug.log): failed to open stream: No such file or directory in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 44
PHP message: PHP Fatal error:  Uncaught exception 'RuntimeException' with message 'Unable to write to the log file.' in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php:45
Stack trace:
#0 /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/Logger.php(76): SimpleLogger\File->log('debug', 'SQL: SELECT "op...', Array)
#1 /srv/kanban/vendor/psr/log/Psr/Log/AbstractLogger.php(126): SimpleLogger\Logger->log('debug', 'SQL: SELECT "op...', Array)
#2 /srv/kanban/app/Subscriber/BootstrapSubscriber.php(32): Psr\Log\AbstractLogger->debug('SQL: SELECT "op...')
#3 [internal function]: Kanboard\Subscriber\BootstrapSubscriber->__destruct()
#4 {main}
  thrown in /srv/kanban/vendor/fguillot/simpleLogger/src/SimpleLogger/File.php on line 45" while reading upstream, client: 10.0.4.9, server: kanban.fedasil.be, request: "POST /?controller=AuthController&action=check HTTP/1.1", upstream: "fastcgi://unix:/var/run/php5-fpm.sock:", host: "kanban.fedasil.be"
```

Oddly enough, it claims it can't write to the debug log file? But there is output in there? www-data has read and write permissions on it. I'll just paste it's contents as proof. Do note that this log seems to jump back and forth between timezones, in case that has anything to do with it.

```
[2017-01-18 09:16:21] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-18 09:16:21] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\AuthController::logout
[2017-01-18 08:16:21] [debug] Subscriber executed: Kanboard\Subscriber\AuthSubscriber::afterLogout
[2017-01-18 09:16:21] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-18 09:16:21] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-18 08:16:21] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\AuthController::login
[2017-01-18 09:16:36] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-18 09:16:36] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-18 08:16:36] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-18 08:16:36] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\AuthController::login
[2017-01-18 09:16:41] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-18 09:16:41] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-18 08:16:41] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-18 08:16:41] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\AuthController::check
```

**Comment by fguillot** [comment]
This is really weird. Do you have only those parameters in your config file?

```php
<?php

define('DEBUG', true);
define('LOG_DRIVER', 'file');
```

I will try to reproduce the problem with a VM and a fresh install of Debian Jessie.

**Comment by rakesh1084** [comment]
Even I getthe sameerror 
    Application version: 1.0.36
    PHP version: 5.3.28
    PHP SAPI: cgi-fcgi
    OS version: Windows NT 6.2
    Database driver: mysql
    Database version: 5.6.31
    Browser: Mozilla/5.0 (Windows NT 6.3; WOW64; rv:47.0) Gecko/20100101 Firefox/47.0


Below is my config file.
```
<?php

/*******************************************************************/
/* Rename this file to config.php if you want to change the values */
/*******************************************************************/

// Data folder (must be writeable by the web server user)
define('DATA_DIR', 'data');

// Enable/Disable debug
define('DEBUG', true);

// Available log drivers: syslog, stderr, stdout or file
define('LOG_DRIVER', 'file');

// Log filename if the log driver is "file"
define('LOG_FILE', DATA_DIR.DIRECTORY_SEPARATOR.'debug.log');

// Plugins directory
define('PLUGINS_DIR', 'plugins');

// Plugins directory URL
define('PLUGIN_API_URL', 'https://kanboard.net/plugins.json');

// Enable/Disable plugin installer
define('PLUGIN_INSTALLER', true);

// Available cache drivers are "file" and "memory"
define('CACHE_DRIVER', 'memory');

// Cache folder to use if cache driver is "file" (must be writeable by the web server user)
define('CACHE_DIR', DATA_DIR.DIRECTORY_SEPARATOR.'cache');

// Folder for uploaded files (must be writeable by the web server user)
define('FILES_DIR', DATA_DIR.DIRECTORY_SEPARATOR.'files');

// E-mail address for the "From" header (notifications)
define('MAIL_FROM', 'rakesh1084@gmail.com');

// Mail transport available: "smtp", "sendmail", "mail" (PHP mail function), "postmark", "mailgun", "sendgrid"
define('MAIL_TRANSPORT', 'smtp');

// SMTP configuration to use when the "smtp" transport is chosen
define('MAIL_SMTP_HOSTNAME', '');
define('MAIL_SMTP_PORT', 25);
define('MAIL_SMTP_USERNAME', '');
define('MAIL_SMTP_PASSWORD', '');
define('MAIL_SMTP_ENCRYPTION', null); // Valid values are "null", "ssl" or "tls"

// Sendmail command to use when the transport is "sendmail"
define('MAIL_SENDMAIL_COMMAND', '/usr/sbin/sendmail -bs');

// Run automatically database migrations
// If set to false, you will have to run manually the SQL migrations from the CLI during the next Kanboard upgrade
// Do not run the migrations from multiple processes at the same time (example: web page + background worker)
define('DB_RUN_MIGRATIONS', true);

// Database driver: sqlite, mysql or postgres (sqlite by default)
define('DB_DRIVER', 'mysql');

// Mysql/Postgres username
define('DB_USERNAME', 'kb_testing');

// Mysql/Postgres password
define('DB_PASSWORD', 'kb_testing');

// Mysql/Postgres hostname
define('DB_HOSTNAME', 'localhost');

// Mysql/Postgres database name
define('DB_NAME', 'kanban_testing');

// Mysql/Postgres custom port (null = default port)
define('DB_PORT', 33061);

// Mysql SSL key
define('DB_SSL_KEY', null);

// Mysql SSL certificate
define('DB_SSL_CERT', null);

// Mysql SSL CA
define('DB_SSL_CA', null);

// Enable LDAP authentication (false by default)
define('LDAP_AUTH', true);

define('LDAP_SERVER', 'x.x.x.x');

// LDAP server port (389 by default)
define('LDAP_PORT', 389);

// By default, require certificate to be verified for ldaps:// style URL. Set to false to skip the verification
define('LDAP_SSL_VERIFY', false);

// Enable LDAP START_TLS
define('LDAP_START_TLS', false);

// By default Kanboard lowercase the ldap username to avoid duplicate users (the database is case sensitive)
// Set to true if you want to preserve the case
define('LDAP_USERNAME_CASE_SENSITIVE', false);

// LDAP bind type: "anonymous", "user" or "proxy"
define('LDAP_BIND_TYPE', 'user');

// LDAP username to use with proxy mode
// LDAP username pattern to use with user mode
define('LDAP_USERNAME', '');

// LDAP password to use for proxy mode
define('LDAP_PASSWORD', '');

// LDAP DN for users
// Example for ActiveDirectory: CN=Users,DC=kanboard,DC=local
// Example for OpenLDAP: ou=People,dc=example,dc=com
define('LDAP_USER_BASE_DN', 'dc=org,dc=COM');

// LDAP pattern to use when searching for a user account
// Example for ActiveDirectory: '(&(objectClass=user)(sAMAccountName=%s))'
// Example for OpenLDAP: 'uid=%s'
define('LDAP_USER_FILTER', '(&(objectClass=user)(sAMAccountName=%s))');

// LDAP attribute for username
// Example for ActiveDirectory: 'samaccountname'
// Example for OpenLDAP: 'uid'
define('LDAP_USER_ATTRIBUTE_USERNAME', 'sAMAccountName');

// LDAP attribute for user full name
// Example for ActiveDirectory: 'displayname'
// Example for OpenLDAP: 'cn'
define('LDAP_USER_ATTRIBUTE_FULLNAME', 'givenName');

// LDAP attribute for user email
define('LDAP_USER_ATTRIBUTE_EMAIL', 'mail');

// LDAP attribute to find groups in user profile
define('LDAP_USER_ATTRIBUTE_GROUPS', 'memberof');

// LDAP attribute for user avatar image: thumbnailPhoto or jpegPhoto
define('LDAP_USER_ATTRIBUTE_PHOTO', 'thumbnailphoto');

// LDAP attribute for user language, example: 'preferredlanguage'
// Put an empty string to disable language sync
define('LDAP_USER_ATTRIBUTE_LANGUAGE', '');

// Allow automatic LDAP user creation
define('LDAP_USER_CREATION', false);

// LDAP DN for administrators
// Example: CN=Kanboard-Admins,CN=Users,DC=kanboard,DC=local
define('LDAP_GROUP_ADMIN_DN', '');

// LDAP DN for managers
// Example: CN=Kanboard Managers,CN=Users,DC=kanboard,DC=local
define('LDAP_GROUP_MANAGER_DN', '');

// Enable LDAP group provider for project permissions
// The end-user will be able to browse LDAP groups from the user interface and allow access to specified projects
define('LDAP_GROUP_PROVIDER', true);

// LDAP Base DN for groups
define('LDAP_GROUP_BASE_DN', 'memberof');

// LDAP group filter
// Example for ActiveDirectory: (&(objectClass=group)(sAMAccountName=%s*))
define('LDAP_GROUP_FILTER', '(&(objectClass=group)(sAMAccountName=%s*)');

// LDAP user group filter
// If this filter is configured, Kanboard will search user groups in LDAP_GROUP_BASE_DN with this filter
// Example for OpenLDAP: (&(objectClass=posixGroup)(memberUid=%s))
define('LDAP_GROUP_USER_FILTER', '(&(objectClass=group)(sAMAccountName=%s*)');

// LDAP attribute for the group name
define('LDAP_GROUP_ATTRIBUTE_NAME', 'groups');

// Enable/disable the reverse proxy authentication
define('REVERSE_PROXY_AUTH', false);

// Header name to use for the username
define('REVERSE_PROXY_USER_HEADER', 'REMOTE_USER');

// Username of the admin, by default blank
define('REVERSE_PROXY_DEFAULT_ADMIN', '');

// Default domain to use for setting the email address
define('REVERSE_PROXY_DEFAULT_DOMAIN', '');

// Enable/disable remember me authentication
define('REMEMBER_ME_AUTH', true);

// Enable or disable "Strict-Transport-Security" HTTP header
define('ENABLE_HSTS', true);

// Enable or disable "X-Frame-Options: DENY" HTTP header
define('ENABLE_XFRAME', true);

// Escape html inside markdown text
define('MARKDOWN_ESCAPE_HTML', true);

// API alternative authentication header, the default is HTTP Basic Authentication defined in RFC2617
define('API_AUTHENTICATION_HEADER', 'X-API-Auth');

// Enable/disable url rewrite
define('ENABLE_URL_REWRITE', true);

// Hide login form, useful if all your users use Google/Github/ReverseProxy authentication
define('HIDE_LOGIN_FORM', false);

// Disabling logout (useful for external SSO authentication)
define('DISABLE_LOGOUT', false);

// Enable captcha after 3 authentication failure
define('BRUTEFORCE_CAPTCHA', 3);

// Lock the account after 6 authentication failure
define('BRUTEFORCE_LOCKDOWN', 6);

// Lock account duration in minute
define('BRUTEFORCE_LOCKDOWN_DURATION', 15);

// Session duration in second (0 = until the browser is closed)
// See http://php.net/manual/en/session.configuration.php#ini.session.cookie-lifetime
define('SESSION_DURATION', 0);

// HTTP client proxy
define('HTTP_PROXY_HOSTNAME', '');
define('HTTP_PROXY_PORT', '3128');
define('HTTP_PROXY_USERNAME', '');
define('HTTP_PROXY_PASSWORD', '');

// Set to false to allow self-signed certificates
define('HTTP_VERIFY_SSL_CERTIFICATE', true);

// TOTP (2FA) issuer name
define('TOTP_ISSUER', 'Kanboard');


```
===============================================================
===============================================================

below are the contents in the new task screen
================================================================


Warning: file_put_contents(data\debug.log) [function.file-put-contents]: failed to open stream: 
No such file or directory in ..\vendor\fguillot\simpleLogger\src\SimpleLogger\File.php on line 44

Fatal error: Uncaught exception 'RuntimeException' with message 'Unable to write to the log file.' in 
..\vendor\fguillot\simpleLogger\src\SimpleLogger\File.php:45 Stack trace: #0 
..\vendor\fguillot\simpleLogger\src\SimpleLogger\Logger.php(76): SimpleLogger\File->log('debug', 'SQL: SELECT `ve...', Array) 
#1 ..\vendor\psr\log\Psr\Log\AbstractLogger.php(118): SimpleLogger\Logger->log('debug', 'SQL: SELECT `ve...', Array) 
#2 ..\app\Subscriber\BootstrapSubscriber.php(32): Psr\Log\AbstractLogger->debug('SQL: SELECT `ve...') 
#3 [internal function]: Kanboard\Subscriber\BootstrapSubscriber->__destruct() 
#4 {main} thrown in ..\vendor\fguillot\simpleLogger\src\SimpleLogger\File.php on line 45

Warning: filemtime() [function.filemtime]: stat failed for p in 
..\app\Helper\AssetHelper.php on line 38
KB DWH
 
  admin
Menu

    Overview Board Calendar List Gantt 

+
Backlog (2)
	
+
Ready (1)
	
+
Work in progress (2)
	
+
Done (1)
	
==================================================================

===================================================================
Below are the contents in the debug.log of data_dir
===================================================================
[2017-01-19 14:32:01] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:01] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:01] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:01] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:01] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:01] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:01] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\BoardAjaxController::check
[2017-01-19 14:32:11] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:11] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:11] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:11] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:11] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:11] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:11] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\BoardAjaxController::check
[2017-01-19 14:32:20] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:20] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\UserAjaxController::status
[2017-01-19 14:32:21] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:21] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:21] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\BoardAjaxController::check
[2017-01-19 14:32:30] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:30] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:30] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:30] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:30] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:30] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:30] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\TaskCreationController::save
[2017-01-19 14:32:31] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:31] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:31] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:31] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:31] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:31] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:31] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\BoardAjaxController::check
[2017-01-19 14:32:41] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:41] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:41] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:41] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:41] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:41] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:41] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\BoardAjaxController::check
[2017-01-19 14:32:51] [debug] Kanboard\Core\Controller\Runner::executeMiddleware
[2017-01-19 14:32:51] [debug] Subscriber executed: Kanboard\Subscriber\BootstrapSubscriber::execute
[2017-01-19 09:02:51] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\AuthenticationMiddleware
[2017-01-19 09:02:51] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\PostAuthenticationMiddleware
[2017-01-19 09:02:51] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ApplicationAuthorizationMiddleware
[2017-01-19 09:02:51] [debug] Kanboard\Core\Controller\BaseMiddleware::next => Kanboard\Middleware\ProjectAuthorizationMiddleware
[2017-01-19 09:02:51] [debug] Kanboard\Core\Controller\Runner::executeController => \Kanboard\Controller\BoardAjaxController::check

===================================================================
![save_error_jan_19_with_debug_true_1](https://cloud.githubusercontent.com/assets/20356619/22100331/32c9c464-de55-11e6-91d4-0492b1893fbf.png)
![save_error_jan_19_with_debug_true_after_save](https://cloud.githubusercontent.com/assets/20356619/22100330/32a8d83a-de55-11e6-8293-c41464a1aaa7.png)

**Comment by SwampFalc** [comment]
I took the default config file, renamed it to config.php and I changed the two debug settings.

So technically the other settings are in the file but should be at their default state.

**Comment by rakesh1084** [comment]
Hi,

Even I have LDAP and it works with debug parameter set to false.
Change 
// Enable/Disable debug
define('DEBUG', true);
to
// Enable/Disable debug
define('DEBUG', false);

it will work...!

However debug will not work.

**Comment by fguillot** [comment]
@SwampFalc I was able to reproduce the issue on Debian Jessie.

Try to change the default value of `DATA_DIR` to set an absolute path instead of a relative one, for example:

```
define('DATA_DIR', __DIR__.DIRECTORY_SEPARATOR.'data');
```

**Comment by SwampFalc** [comment]
I put in the absolute path on the data folder on my server and that solved the issue.

I also tried my ldap config again (since that was the original problem I had) and it works now. I don't believe I put in anything different from last time. So right now, gut feeling would be that this data dir setting seems to break all logging in.

So for me this issue is closed but I'll leave it to you whether you want to keep this open for the bigger bug.

**Comment by fguillot** [comment]
It should be fixed in v1.0.38
